gobblin-utility/src/main/java/gobblin/util/JobLauncherUtils.java

```diff
@@ -71,27 +71,30 @@ public static String newMultiTaskId(String jobId, int sequence) {
   /**
    * Cleanup staging data of a Gobblin task.
    *
-   * @param taskState task state
+   * @param state workunit state
    */
-  public static void cleanStagingData(State taskState, Logger logger) throws IOException {
-    int branches = taskState.getPropAsInt(ConfigurationKeys.FORK_BRANCHES_KEY, 1);
+  public static void cleanStagingData(WorkUnitState state, Logger logger) throws IOException {
+    int branches = state.getPropAsInt(ConfigurationKeys.FORK_BRANCHES_KEY, 1);
     for (int i = 0; i < branches; i++) {
-      String writerFsUri = taskState
+      String writerFsUri = state
           .getProp(ForkOperatorUtils.getPropertyNameForBranch(ConfigurationKeys.WRITER_FILE_SYSTEM_URI, branches, i),
               ConfigurationKeys.LOCAL_FS_URI);
       FileSystem fs = FileSystem.get(URI.create(writerFsUri), new Configuration());
 
-      String writerFilePath = taskState
+      String writerFilePath = state
           .getProp(ForkOperatorUtils.getPropertyNameForBranch(ConfigurationKeys.WRITER_FILE_PATH, branches, i));
+
+      // If the WRITER_FILE_PATH has not been set, use the default value
       if (Strings.isNullOrEmpty(writerFilePath)) {
-        // The job may be cancelled before the task starts, so this may not be set.
-        continue;
+        writerFilePath =
+            state.getExtract().getOutputFilePath() + Path.SEPARATOR
+                + ForkOperatorUtils.getBranchName(state, i, ConfigurationKeys.DEFAULT_FORK_BRANCH_NAME + i);
       }
 
       String stagingDirKey =
           ForkOperatorUtils.getPropertyNameForBranch(ConfigurationKeys.WRITER_STAGING_DIR, branches, i);
-      if (taskState.contains(stagingDirKey)) {
-        Path stagingPath = new Path(taskState.getProp(stagingDirKey), writerFilePath);
+      if (state.contains(stagingDirKey)) {
+        Path stagingPath = new Path(state.getProp(stagingDirKey), writerFilePath);
         if (fs.exists(stagingPath)) {
           logger.info("Cleaning up staging directory " + stagingPath.toUri().getPath());
           fs.delete(stagingPath, true);
```

Comment:
Is it possible to delete the entire extract staging directory? There are a lot of HDFS operations here and anyone can throw an exception which will leave the staging directory in an inconsistent state. Ideally, that directory is removed in the beginning of the job and the job should fail if the clean up of that directory fails.